test/exe/signals_test.cc

```diff
@@ -0,0 +1,96 @@
+#include <signal.h>
+#include <sys/mman.h>
+
+#include "exe/signal_action.h"
+
+#include "gtest/gtest.h"
+
+#if defined(__has_feature)
+#if __has_feature(address_sanitizer)
+#define ASANITIZED /* Sanitized by Clang */
+#endif
+#endif
+
+#if defined(__SANITIZE_ADDRESS__)
+#define ASANITIZED /* Sanitized by GCC */
+#endif
+
+// Memory violation signal tests are disabled under address sanitizer.  The
+// sanitizer does its own special signal handling and prints messages that are
+// not ours instead of what this test expects.  The signals special handled by ASAN
+// include SIGSEGV, SIGBUS, and SIGFPE.
+#ifndef ASANITIZED
+TEST(Signals, InvalidAddressDeathTest) {
+  SignalAction actions;
+  EXPECT_DEATH([]() -> void {
```

Comment:
You could make these tests much stronger by implementing the behavior we discussed, where you call the original signal handler after dumping the stack trace and installing a signal handler before the definitions of `actions`. That way you could verify that the stack trace dump signal handler didn't crash itself, and possibly that it ran if you set some global state.

Anyway, it's probably overkill for this kind of feature (you already have done a lot of test stuff), but just an FYI.